src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -9580,7 +8715,7 @@ internal bool IsSafeVariableReference(HashSet<string> validVariables, ref bool u
             return ok;
         }
 
-        private bool AstAssignsToSameVariable(Ast ast)
+        internal bool AstAssignsToSameVariable(Ast ast)
         {
```

Comment:
The only reference to this was in `GetInferredType` - so maybe move this method too.